File: frontend/src/types/reviewStats.ts
export interface AgentFeedbackStats {
  agent: string;
  total_risks: number;
  helpful: number;
  not_helpful: number;
  false_positive: number;
  false_positive_rate: number;
}

export interface RecentFalsePositive {
  record_id: number;
  file: string;
  issue: string;
  created_at: string;
}

export interface ReviewStatsResponse {
  total_analyses: number;
  total_risks_flagged: number;
  feedback_coverage: number;
  by_agent: AgentFeedbackStats[];
  false_positive_rate: number;
  recent_false_positives: RecentFalsePositive[];
}

File: backend/app/schemas/review_stats.py
from pydantic import BaseModel


class AgentFeedbackStats(BaseModel):
    agent: str
    total_risks: int = 0
    helpful: int = 0
    not_helpful: int = 0
    false_positive: int = 0
    false_positive_rate: float = 0.0


class RecentFalsePositive(BaseModel):
    record_id: int
    file: str
    issue: str
    created_at: str


class ReviewStatsResponse(BaseModel):
    total_analyses: int = 0
    total_risks_flagged: int = 0
    feedback_coverage: float = 0.0  # ratio: feedback_count / total_risks
    by_agent: list[AgentFeedbackStats] = []
    false_positive_rate: float = 0.0
    recent_false_positives: list[RecentFalsePositive] = []
